src/overlays/actors/ovl_En_Po_Fusen/z_en_po_fusen.c

```diff
@@ -21,34 +29,232 @@ const ActorInit En_Po_Fusen_InitVars = {
     (ActorFunc)EnPoFusen_Update,
     (ActorFunc)EnPoFusen_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Po_Fusen_0x80AC1270/EnPoFusen_Init.asm")
+u32 EnPoFusenSphereInit[] = { //ColSphereInit  
+    0x0A000939, 0x12040000, 0x00000000, 0xF7CFFFFF,
+    0x00000000, 0xF7CFFFFF, 0x00000000, 0x00010100,
+    0x00000000, 0x00000000, 0x00C80064, };
+
+DamageTable EnPoFusenDamageTable = { 
+    0x00, 0x00, 0x00, 0x00, 0x00, 0xF1, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF1, 0xF1, 0xF1, 0xF1, 0x00, 
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 
+};
+
+u32 unkpadding = 0x0;
+
+void EnPoFusen_Init(Actor *thisx, GlobalContext *globalCtx) {
+    EnPoFusen* this = THIS;
+    f32 heightTemp;
+
+    this->actor.scale.x = this->actor.scale.y = this->actor.scale.z = 0.00700000021607;
+    this->actor.targetMode = 6;
+    this->actor.colChkInfo.damageTable = &EnPoFusenDamageTable;
+
+    Collision_InitSphereDefault(globalCtx, &this->collider);
+    Collision_InitSphereWithData(globalCtx, &this->collider, &this->actor, &EnPoFusenSphereInit);
+    if (0){}
+    this->collider.params.colInfo.radius  = 40;
+    SkelAnime_InitSV(globalCtx, &this->anime, &D_060024F0, &D_06000040,
+        &this->limbDrawTbl, &this->transitionDrawTbl, 10);
+    Actor_SetDrawParams(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    func_800B78B8(globalCtx, this, 0.0f, 0.0f, 0.0f, 4);
+    if (EnPoFusen_CheckParent(this, globalCtx) == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+
+    heightTemp = this->actor.floorHeight + 90.0f;
+    if (this->actor.home.pos.y < heightTemp) {
+        this->actor.home.pos.y = heightTemp;
+    }
+
+    this->randScaleChange = (s16) ((rand() % 0xFFFEU) - 0x7FFF);
+    this->randYRotChange = (s16) ((rand() % 0x4B0U) - 600);
+    this->avgBaseRotation = 5461;
+    this->limb3Rot = 0;
+    this->limb46Rot = 0;
+    this->limb57Rot = 0;
+    this->limb8Rot = 0;
+    this->limb9Rot = 1820;
+    this->randBaseRotChange = 0;
+    if ((this->actor.params & 0x8000) != 0) {
+        EnPoFusen_InitFuse(this);
+        return;
+    }
+    EnPoFusen_Init2(this);
+}
+
+void EnPoFusen_Destroy(Actor* thisx, GlobalContext *gCtx) {
+    EnPoFusen* this = THIS;
+    Collision_FiniSphere(gCtx, &this->collider);
+}
+
+u16 EnPoFusen_CheckParent(EnPoFusen *this, GlobalContext *globalCtx) {
+    struct Actor *actorPtr;
+
+    actorPtr = globalCtx->actorCtx.actorList[4].first;
+    if ((this->actor.params & 0x8000) != 0) {
```

Comment:
Since you've done documentation on this actor already, I would suggest making macros for read/write access to these parameters.